src/extension.ts

```diff
@@ -90,24 +123,28 @@ export function activate(context: vscode.ExtensionContext) {
         return positions;
     }
 
-    function readConfiguration(): { mode: string, regex: RegExp } {
-        const ligatureConfiguration = vscode.workspace.getConfiguration().get("disableLigatures") as any;
-        const ligatures = ligatureConfiguration.ligatures as string[];
+    function readConfiguration(): { mode: string, regex: RegExp, globalRegex: RegExp } {
+        const ligatureConfiguration = Object(vscode.workspace.getConfiguration().get("disableLigatures")) as any;
+        const globalLigatures = Array.from(ligatureConfiguration.globalLigatures || []) as string[];
+        const ligatures = Array.from(ligatureConfiguration.ligatures || []) as string[];
+        const sortByLongestFirst = (a: any, b: any) => b.length - a.length;
 
         // Match longest first
-        ligatures.sort((a, b) => {
-            return b.length - a.length;
-        });
+        globalLigatures.sort(sortByLongestFirst);
+        ligatures.sort(sortByLongestFirst);
 
         const escapeRegex = /[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g;
 
         // tslint:disable-next-line:prefer-for-of
         for (let i = 0; i < ligatures.length; i++) {
+            globalLigatures[i] = globalLigatures[i].replace(escapeRegex, "\\$&");
```

Comment:
This errored out for me, since the length is taken from ligatures but applies to globalLigatures

Reply:
Yea there should probably be two different loops for this